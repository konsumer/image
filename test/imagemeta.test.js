import { readFile } from 'node:fs/promises'
import { info, infoFetch } from '../src/index.js'

test('sus-palette-a.png', async () => {
  const image = await readFile('test/sus-palette-a.png')
  expect(info(image)).toMatchSnapshot()
})

test('sus-palette-no_a.png', async () => {
  const image = await readFile('test/sus-palette-no_a.png')
  expect(info(image)).toMatchSnapshot()
})

test('sus-rgb-a.png', async () => {
  const image = await readFile('test/sus-rgb-a.png')
  expect(info(image)).toMatchSnapshot()
})

test('sus-rgb-no_a.png', async () => {
  const image = await readFile('test/sus-rgb-no_a.png')
  expect(info(image)).toMatchSnapshot()
})

test('sus.jpg', async () => {
  const image = await readFile('test/sus.jpg')
  expect(info(image)).toMatchSnapshot()
})

test('remote jpg', async () => {
  expect(await infoFetch('https://placebear.com/200/300.jpg')).toMatchSnapshot()
})

test('remote png', async () => {
  expect(await infoFetch('https://sample-videos.com/img/Sample-png-image-100kb.png')).toMatchSnapshot()
})
