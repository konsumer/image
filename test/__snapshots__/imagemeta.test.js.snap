// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`remote jpg 1`] = `
{
  "bitDepth": 8,
  "components": 3,
  "height": 300,
  "progressive": false,
  "type": "jpg",
  "width": 200,
}
`;

exports[`remote png 1`] = `
{
  "alpha": true,
  "bitDepth": 8,
  "colorType": 6,
  "compressionMethod": 0,
  "filterMethod": 0,
  "height": 170,
  "interlaceMethod": 0,
  "type": "png",
  "width": 272,
}
`;

exports[`sus.jpg 1`] = `
{
  "bitDepth": 8,
  "components": 3,
  "height": 1024,
  "progressive": true,
  "type": "jpg",
  "width": 842,
}
`;

exports[`sus-palette-a.png 1`] = `
{
  "alpha": true,
  "bitDepth": 8,
  "colorType": 3,
  "compressionMethod": 0,
  "filterMethod": 0,
  "height": 1024,
  "interlaceMethod": 0,
  "type": "png",
  "width": 842,
}
`;

exports[`sus-palette-no_a.png 1`] = `
{
  "alpha": false,
  "bitDepth": 8,
  "colorType": 3,
  "compressionMethod": 0,
  "filterMethod": 0,
  "height": 1024,
  "interlaceMethod": 0,
  "type": "png",
  "width": 842,
}
`;

exports[`sus-rgb-a.png 1`] = `
{
  "alpha": true,
  "bitDepth": 16,
  "colorType": 6,
  "compressionMethod": 0,
  "filterMethod": 0,
  "height": 1024,
  "interlaceMethod": 0,
  "type": "png",
  "width": 842,
}
`;

exports[`sus-rgb-no_a.png 1`] = `
{
  "alpha": false,
  "bitDepth": 8,
  "colorType": 2,
  "compressionMethod": 0,
  "filterMethod": 0,
  "height": 1024,
  "interlaceMethod": 0,
  "type": "png",
  "width": 842,
}
`;
